app/controllers/finders_controller.rb

```diff
@@ -2,10 +2,10 @@ class FindersController < ApplicationController
   layout "finder_layout"
 
   before_action do
-    expires_in(5.minutes, public: true)
+    content_item = ContentItem.from_content_store(finder_base_path)
```

Comment:
In the worst case (the cache is unavailable) we're calling the content store twice. In the best case (cache is warm) we're calling the cache twice. I think we can do a bit better and have a memoized method (e.g. def content_store; @content_item ||= ContentItem.from_content_store(finder_base_path); end`) so we don't call it twice.